apps/homework/tasks.py

```diff
@@ -79,60 +102,68 @@ def post_submission(submission_pk, data_file=None):
             data_to_upload = data_file
             temp_size = str(data_to_upload.size)
         if data_to_upload:
-            storage_resp = requests.put(
-                url=s3_file_url,
-                data=data_to_upload,
-                headers={
-                    "x-ms-blob-type": 'BlockBlob',
-                    "Content-Length": temp_size,
-                }
-            )
+            if submission.definition.jupyter_notebook_enabled:
+                # Temporary files can only be opened once, so both reads need to happen within this context manager
+                with data_to_upload.open() as fd:
+                    numerator, denominator = retrieve_score_from_jupyter_notebook(data_to_upload)
+                    fd.seek(0)
```

Comment:
maybe explain why we're seeking back to 0 for re-reading the file

Reply:
I thought the comment above might be clear, but I'll add some more detail.✅

Reply:
ty